ee.go

```diff
@@ -5,23 +5,32 @@ import (
 	"fmt"
 )
 
-type EventHandler = func(data ...any)
+type EventHandlerFunc = func(data ...any)
+
+type EventHandler struct {
+	Id   int
+	Func EventHandlerFunc
+}
 
 type EventEmitter struct {
 	events map[string][]EventHandler
 }
 
 // On subscribes handler to event and returns index of handler
 // It also creates internal events map if struct wasn't initialized
-func (emitter *EventEmitter) On(event string, handler EventHandler) int {
+func (emitter *EventEmitter) On(event string, handlerFunc EventHandlerFunc) int {
 	if emitter.events == nil {
 		emitter.events = make(map[string][]EventHandler)
 	}
 
+	handler := EventHandler{
+		Id:   len(emitter.events[event]) - 1,
```

Comment:
TODO: change way to generate ids, cause rn there is a chance there will be several handlers with same id.

for example, we can add 2 handlers, remove first one, and add another 3.